adapters/sharethrough/sharethrough.go

```diff
@@ -137,6 +137,9 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 			if bidReq.Imp[0].Video != nil {
 				bidType = openrtb_ext.BidTypeVideo
 			}
+			if bidReq.Imp[0].Native != nil {
```

Comment:
Warning:- we recommend setting the bidType based on bidder response. Your bidder should be able to set seatBid.Bid.Ext.Prebid.Type field and use that field here to get bidType.

Reply:
Agreed with above comment. 
This is an anti-pattern. This code assumes that if there is multi-format request then mediatype is defaulting to native. Ideally, we would like to set media type based on adapter response. `bidresponse.seatbid.bid[i].ext.prebid.type` field identifies mediaType. So, your adapter should set this field in its response to define media type.

Reply:
Thank you for the great feedback @Sonali-More-Xandr @gargcreation1992 .  Updates in new commit (`96c978f`).